ext/js/pages/settings/dictionary-controller.js

```diff
@@ -113,6 +122,36 @@ class DictionaryEntry {
         this._enabledCheckbox.checked = value;
     }
 
+    /**
+     * @returns {Promise<boolean>}
+     */
+    async checkForUpdate() {
+        this._updatesAvailable.hidden = true;
+        const {isUpdatable, indexUrl, revision: currentRevision, downloadUrl: currentDownloadUrl} = this._dictionaryInfo;
+        if (!isUpdatable || !indexUrl || !currentDownloadUrl) { return false; }
```

Comment:
if I'm understanding correctly, `currentDownloadUrl` in the existing dictionary is required? I feel like this is unnecessary as long as the new index has one.

Reply:
Depending on workflow it may redundant, but I think it doesn't interfere with any use style. If you use a different `downloadUrl` every time, it will get the one from the new index.